cpp/ql/src/semmle/code/cpp/models/implementations/StdString.qll

```diff
@@ -58,6 +65,35 @@ class StdStringAppend extends TaintFunction {
     getParameter(result).getType() = getDeclaringType().getTemplateArgument(0) // i.e. `std::basic_string::CharT`
   }
 
+  override predicate hasTaintFlow(FunctionInput input, FunctionOutput output) {
+    // flow from string and parameter to string (qualifier) and return value
+    (
+      input.isQualifierObject() or
+      input.isParameterDeref(getAStringParameter())
```

Comment:
Will this correctly classify taint from iterators? i.e., will we get taint to `s2` in the following case:
```cpp
std::string s = source();
std::string s2;
s2.append(s.begin(), s.end());
```
or do we need to extend this predicate once we have a model for iterators?

Reply:
The intent for this particular PR is not to model taint through the iterator versions of the functions.  I wanted to see what's decided / done for https://github.com/github/codeql-c-analysis-team/issues/122 before addressing them.